addon/index.js

```diff
@@ -25,8 +25,19 @@ window.addEventListener('storage', function ({ key, newValue }) {
   localStorageCache.set(key, jsonParseAndFreeze(newValue));
 });
 
-export default function localStorageDecorator(customLocalStorageKey) {
-  return function (target, key, descriptor) {
+export default function localStorageDecoratorFactory(...args) {
+  const isDirectDecoratorInvocation = isElementDescriptor(...args);
+  let customLocalStorageKey = null;
```

Comment:
I think this could be refactored to a const variable for readability.

I would also prefer to have it either `undefined` or a string. Using `null` for direct decorator invocation increases complexity in my opinion.

```suggestion
  const customLocalStorageKey = isDirectDecoratorInvocation ? undefined : args[0];
```